core/unit_test/cuda/TestCuda_TeamScratch.cpp

```diff
@@ -47,22 +47,121 @@
 
 namespace Test {
 
+namespace Impl {
+
+struct CudaStreamScratchTestFunctor {
+  using team_t    = Kokkos::TeamPolicy<Kokkos::Cuda>::member_type;
+  using scratch_t = Kokkos::View<int64_t*, Kokkos::Cuda::scratch_memory_space>;
+
+  Kokkos::View<int64_t, Kokkos::CudaSpace, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      counter;
+  int N, M;
+  CudaStreamScratchTestFunctor(
+      Kokkos::View<int64_t, Kokkos::CudaSpace> counter_, int N_, int M_)
+      : counter(counter_), N(N_), M(M_) {}
+
+  KOKKOS_FUNCTION
+  void operator()(const team_t& team) const {
+    scratch_t scr(team.team_scratch(1), M);
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M),
+                         [&](int i) { scr[i] = 0; });
+    team.team_barrier();
+    for (int i = 0; i < N; i++) {
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M),
+                           [&](int j) { scr[j] += 1; });
+    }
+    team.team_barrier();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, 0, M), [&](int i) {
+      if (scr[i] != N) counter()++;
+    });
+  }
+};
+
+void cuda_stream_scratch_test_one(
+    int N, int T, int M_base, Kokkos::View<int64_t, Kokkos::CudaSpace> counter,
+    Kokkos::Cuda cuda, int tid) {
+  int M = M_base + tid * 5;
+  Kokkos::TeamPolicy<Kokkos::Cuda> p(cuda, T, 64);
+  using team_t    = Kokkos::TeamPolicy<Kokkos::Cuda>::member_type;
```

Comment:
```suggestion
```
ClangTidy build complains that it is not use.
All other CUDA builds fail.  There is an illegal memory address somewhere.